core/blockchain.go

```diff
@@ -1959,6 +1959,26 @@ func (bc *BlockChain) WriteEpochVdfBlockNum(epoch *big.Int, blockNum *big.Int) e
 	return nil
 }
 
+// WriteCrossLinks saves the hashes of crosslinks by shardID and blockNum combination key
+func (bc *BlockChain) WriteCrossLinks(cls []types.CrossLink) error {
+	var err error
+	for _, cl := range cls {
+		err = rawdb.WriteCrossLinkShardBlock(bc.db, cl.ShardID(), cl.BlockNum(), cl.Bytes())
```

Comment:
cross-link should only happen on shard 0

Reply:
shardID here is from other shardID, only shard0 will call this function to write into datebase